tools/installed-map09-refactor-dogfood.php

```
@@ -0,0 +1,771 @@
+        if ($this->scenario === 'method') {
+            $greeter = (string) file_get_contents($this->consumer . '/src/Greeter.php');
+            $caller = (string) file_get_contents($this->consumer . '/src/Caller.php');
+            if (str_contains($greeter . $caller, 'oldName') || !str_contains($greeter . $caller, 'renamedMethod')) {
+                throw new InstalledMap09DogfoodFailure('Method rename did not rewrite exact source scope.');
+            }
+            return;
+        }
+        if ($this->scenario === 'parameter') {
+            $greeter = (string) file_get_contents($this->consumer . '/src/Greeter.php');
+            if (!str_contains($greeter, 'format(string $person)')
+                || !str_contains($greeter, "'Hello ' . $person")
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$person'. (undefined)

(UnusedLocalVariable)